### ISSUE TITLE
Unnecessary Compiler rebuilds 

### ISSUE THREAD

**Issue by rmarinho** [xamarin/xamarin-android#8306]
---
**Issue moved from dotnet/maui#17005**
- Please respond to @genifycom.

---

_From @genifycom on Friday, August 25, 2023 4:20:07 PM_

### Description

If you are working in DotNet MAUI release mode on a physical Android device, the build process has to rebuild 188+ libraries every time to ".so" whenever you make a change. This makes testing the release build unnecessarily lengthy.

Since the 188 core .Net files do not change during this testing cycle, it seems unreasonable to have to wait for a conversion to ".so" for each of these files every time.

Thanks 

### Steps to Reproduce

Create an Android release and run on a physical Android 33 device.

Deploy, then make a source code change and redeploy.

### Version with bug

7.0.92

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 13

### Did you find any workaround?

None

### Relevant log output

```shell
None
```

**Comment by jpobst** [comment]
We definitely optimize the Debug build for the "inner loop" deployment process so that developing your app is quicker.  The Release build is optimized for squeezing every last bit of performance and size out of your app, and is expected to take much, much longer.  So this is largely expected.

Note that in Release mode, linking/trimming is running, so changing _**your**_ code absolutely can change the core .NET libraries since they are modified to only contain the code you need.

Having said that, there may be some additional places where we can detect that things didn't actually change and avoid some unnecessary work.

**Comment by jonathanpeppers** [comment]
Yes, the trimmer & AOT compilers are not incremental at all. You could use a new BCL type and all of a sudden, every assembly changes.

**Comment by jonpryor** [comment]
Closing as "not a bug".  "Fast Release builds" are absolutely *not* a design goal, for all the reasons that @jpobst and @jonathanpeppers have previously mentioned.  If you want fast builds, use the Debug configuration, which *is* intended to be "fast" (and skips a great number of things to be fast).
